server_addon/maya/client/ayon_maya/plugins/publish/collect_ornatrix_rig.py

```diff
@@ -0,0 +1,85 @@
+import os
+from typing import List, Dict, Any
+import pyblish.api
+from ayon_core.pipeline.publish import KnownPublishError
+from ayon_maya.api import lib
+from ayon_maya.api import plugin
+from maya import cmds
+
+
+ORNATRIX_NODES = {
+    "HairFromGuidesNode", "GuidesFromMeshNode",
+    "MeshFromStrandsNode", "SurfaceCombNode"
+}
+
+
+class CollectOxRig(plugin.MayaInstancePlugin):
+    """Collect all information of the Ornatrix Rig"""
+
+    order = pyblish.api.CollectorOrder + 0.4
+    label = "Collect Ornatrix Rig"
+    families = ["oxrig"]
+
+    def process(self, instance):
+        assert "input_SET" in instance.data["setMembers"], (
+            "Ornatrix Rig must have an input_SET")
+
+        ornatrix_nodes = cmds.ls(instance[:], long=True)
+        self.log.debug(f"Getting ornatrix nodes: {ornatrix_nodes}")
+        # Force frame range for yeti cache export for the rig
+        # Collect any textures if used
+        ornatrix_resources = []
+        for node in ornatrix_nodes:
+            # Get Yeti resources (textures)
+            resources = self.get_texture_resources(node)
+            ornatrix_resources.extend(resources)
+        # avoid duplicate dictionary data
+        instance.data["resources"] = [
+            i for n, i in enumerate(ornatrix_resources)
+            if i not in ornatrix_resources[n + 1:]
+        ]
+        self.log.debug("{}".format(instance.data["resources"]))
+
+    def get_texture_resources(self, node: str) -> List[Dict[str, Any]]:
+        resources = []
+        node_shape = cmds.listRelatives(node, shapes=True)
+        if not node_shape:
+            return []
+
+        ox_nodes = [
+            ox_node for ox_node in cmds.listConnections(node_shape, destination=True)
+                    if cmds.nodeType(ox_node) in ORNATRIX_NODES
+        ] or []
```

Comment:
This should be the same as:
```suggestion
        ox_nodes = cmds.listConnections(
            node_shape, destination=True, type=ORNATRIX_NODES) or []
```

Reply:
got the error below if I choose to changing this:
```   
TypeError: Invalid arguments for flag 'type'.  Expected string, got [ str, str, str, str ]
```

Reply:
Interesting. Oh, funny maya. In that case:
```suggestion
        ox_nodes = cmds.ls(cmds.listConnections(
            node_shape, destination=True) or [], type=ORNATRIX_NODES)
```

Reply:
this one works and I committed the changes in [fe2d422](https://github.com/ynput/ayon-core/pull/651/commits/fe2d42299b94eb685a09573e9349418d7fc2638b)